GoMeet/src/main/java/com/google/sps/servlets/MeetingTimeServlet.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    // filter by Key
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    FilterPredicate filter 
+        = new FilterPredicate(MeetingTimeFields.MEETING_TIME_ID, FilterOperator.EQUAL, keyStr);
+    Query query = new Query("MeetingTime").setFilter(filter);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    Entity result;
+    // Expecting only one entity to be returned, since keys should be unique
+    try {
+      result = preparedQuery.asSingleEntity();
+    } catch (PreparedQuery.TooManyResultsException e) {
+      // there is more than the expected number of entities returned - non-unique key
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_REQUEST_ENTITY_TOO_LARGE, ErrorMessages.TOO_MANY_RESULTS_ERROR);
+      return;
+    }
+
+    if (result == null) {
+      // entity by the given key is not found
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_NOT_FOUND, ErrorMessages.ENTITY_NOT_FOUND_ERROR);
+      return;
+    }
+    
+    // Each of the fields will be set to null if not found in returned entity
+    String datetime = (String) result.getProperty(MeetingTimeFields.DATETIME);
+    Long voteCount = 
+        result.getProperty(MeetingTimeFields.VOTE_COUNT) == null ? 
+        null : (long) result.getProperty(MeetingTimeFields.VOTE_COUNT);
+    List voters = (ArrayList) result.getProperty(MeetingTimeFields.VOTERS);
+    // add all to hashmap,
+    HashMap<String, Object> meetingTime = new HashMap<String, Object>() {{
+      put(MeetingTimeFields.DATETIME, datetime);
+      put(MeetingTimeFields.VOTE_COUNT, voteCount);
+      put(MeetingTimeFields.VOTERS, voters);
+    }};
+
+    // return as JSON
+    String meetingTimeJson = ServletUtil.convertToJson(meetingTime);
+    response.getWriter().println(meetingTimeJson);
+  }
+  
+  /** Creates a new MeetingTime entity and stores it to Datastore */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // expect the query string to be of the following format: ?datetime=DATETIME
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String datetime = request.getParameter(MeetingTimeFields.DATETIME);
+
+    if (datetime == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.BAD_REQUEST_ERROR);
+      return;
+    }
+
+    Entity meetingTime = new Entity("MeetingTime");
+    Key meetingTimeKey = meetingTime.getKey();
+    meetingTime.setProperty(MeetingTimeFields.MEETING_TIME_ID, meetingTimeKey.toString());
+    meetingTime.setProperty(MeetingTimeFields.DATETIME, datetime);
+    meetingTime.setProperty(MeetingTimeFields.VOTE_COUNT, 0); // initially votes are 0
+    datastore.put(meetingTime);
+
+    // return the key of the created entity
+    response.setContentType("application/json");
+    response.getWriter().println(meetingTimeKey.toString());
```

Comment:
Consider using KeyFactory.keyToString() instead, which converts the Key into a web-safe string. key.toString() is more suitable for debugging and logging. 

https://cloud.google.com/appengine/docs/standard/java/datastore/entities

Reply:
Oh, okay, didn't know this. Thanks for pointing it out! 😊 Fixing it right now!